### PR TITLE
add tags_diff()

### LINKED ISSUE #325: tags_diff function
# Summary

As a user, I would like to be able to easily see the difference in the `tags` element of different models. I would like to be able to either compare two models directly, or compare all models in a `bbi_run_log_df` to their  "parent models" (a.k.a. the models in that same tibble that match their `based_on`).

# Tests
- tests/testthat/test-tags-diff.R
  - tags_diff.bbi_model default happy path works
  - tags_diff.bbi_model print works
  - tags_diff.bbi_model .mod2 works
  - tags_diff.bbi_run_log_df works

### PR DESCRIPTION
**Pull request by seth127**
closes #325

### CODE REVIEW

**Comment by seth127** [comment]
@timwaterhouse @callistosp this is ready for you to check out and play with. Most of what I did is exactly what we talked about, but I did make one notable change, which I describe [here](https://github.com/metrumresearchgroup/bbr/issues/325#issuecomment-786409395). Enjoy.

I tagged it so you can pull it into whatever project to try it out by adding this to `pkgr.yml`
```
Repos:
  - bbr: https://s3.amazonaws.com/mpn.metworx.dev/releases/bbr/1.0.0.8000.1
```
(though sometimes `pkgr` doesn't recognize those extra versions at the end as needing to update, so you need to actually go delete the `bbr` folder from inside `renv/Library/...` first before you run `pkgr install --update`)

File: R/tags-diff.R [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#' * `tags_removed` contains any tags that are on at least one of the models it
+#'   is based on, but _not_ on the relevant model.
+#'
+#' @param .bbi_object The object to compare. Could be a
+#'   `bbi_{.model_type}_model` object or a
+#'   tibble of class `bbi_run_log_df`.
+#' @param .mod2 If a `bbi_model` object is passed, compare `.bbi_object` to
+#'   `.mod2`. If `.mod2 = NULL`, the default, compare the `.bbi_object` model to
+#'   any models in its `based_on` field. **Only valid for `bbi_model`
+#'   method.**
```
Comment (timwaterhouse) [inline R/tags-diff.R:39]: Add that `.mod2` is also a `bbi_{.model_type}_model` object?

File: R/tags-diff.R [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      .m <- read_model(.b)
+      .m[[YAML_TAGS]]
+    }) %>%
+      unlist() %>%
+      unique()
+  }
+
+  # compare tags, print, and return
+  mod_diff     <- setdiff(mod_tags, compare_tags) %||% ""
+  compare_diff <- setdiff(compare_tags, mod_tags) %||% ""
```
Comment (timwaterhouse) [inline R/tags-diff.R:75]: This is nifty

**Review by timwaterhouse — DISMISSED** [review]
Looks good, thanks Seth. I haven't had a chance to try it out yet, but I will.

**Review by callistosp — APPROVED** [review]
LGTM; however, I wasn't able to test on a live repo either

**Comment by callistosp** [comment]
confirmed new functions work in my current project summary script with @seth127's help this morning

File: R/tags-diff.R (hunk 1 above)
Comment (seth127) [reply R/tags-diff.R:39]: I have 
> If a `bbi_model` object is passed...
in there because this will work for Stan models, etc. too

**Comment by seth127** [comment]
Merging this because @timwaterhouse and @callistosp already approved ([here](https://github.com/metrumresearchgroup/bbr/pull/337#pullrequestreview-599869095) and [here](https://github.com/metrumresearchgroup/bbr/pull/337#issuecomment-788038917)) and the only changes that made their reviews "stale" were conflicts in DESCRIPTION

File: R/tags-diff.R (hunk 1 above)
Comment (seth127) [reply R/tags-diff.R:39]: actually, you're right, it should be `bbi_{.model_type}_model`. I'll change it in [the `model_diff` PR](https://github.com/metrumresearchgroup/bbr/pull/341)

**Pull request merged**